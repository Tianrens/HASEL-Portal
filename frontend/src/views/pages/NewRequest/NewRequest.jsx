import { React, useState } from 'react';
import TextField from '@material-ui/core/TextField';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import styles from './NewRequest.module.scss';
import { StyledButton } from '../../components/buttons/StyledButton';

const NewRequest = () => {
    const [supervisor, setSupervisor] = useState('');
    const [reason, setReason] = useState('');
    const [workstation, setWorkstation] = useState('');

    const workstationNames = ['Zeus', 'Apollo', 'Athena'];

    const showSupervisor = false;

    const handleSubmit = (event) => {
        event.preventDefault();
        // send email
        // call API
        console.log(supervisor, reason, workstation);
    };

    return (
        <div className={styles.container}>
            <h1 className={styles.header}>You&apos;ll need approval to access a workstation.</h1>
            <form autoComplete='off' className={styles.form} onSubmit={handleSubmit}>
                {showSupervisor && (
                    <>
                        <p className={styles.inputTitles}>Supervisor name</p>
                        <TextField
                            required
                            fullWidth
                            variant='outlined'
                            onChange={(e) => setSupervisor(e.target.value)}
                        />
                    </>
                )}

                <p className={styles.inputTitles}>Reasoning/Comments (optional)</p>
                <TextField
                    variant='outlined'
                    fullWidth
                    multiline
                    rows={4}
                    onChange={(e) => setReason(e.target.value)}
                />

                <p className={styles.inputTitles}>Workstation (may be overridden)</p>
                <Select
                    required
                    defaultValue=''
                    fullWidth
                    variant='outlined'
                    MenuProps={{
                        anchorOrigin: {
                            vertical: 'bottom',
                            horizontal: 'left',
                        },
                        getContentAnchorEl: null,
                    }}
                    onChange={(e) => setWorkstation(e.target.value)}
                >
                    {workstationNames.map((option) => (
                        <MenuItem key={option} value={option}>
                            {option}
                        </MenuItem>
                    ))}
                </Select>
                <div className={styles.buttonContainer}>
                    <StyledButton className={styles.button} type='submit'>
                        Request Access
                    </StyledButton>
                </div>
            </form>
        </div>
    );
};

export default NewRequest;
