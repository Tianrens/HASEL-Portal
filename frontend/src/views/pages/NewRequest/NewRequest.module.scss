@import '../../../assets/colours';

.header {
    color: $heading;
    text-align: center;
}

.inputTitles {
    color: $body;
}

.inputTitles:not(:first-child) {
    margin-top: 3em;
}

.container {
    display: flex;
    flex-direction: column;
    align-items: center;
}
.form {
    min-width: 40ch;
}

.buttonContainer {
    display: flex;
    justify-content: center;
    margin-top: 3rem;
}
