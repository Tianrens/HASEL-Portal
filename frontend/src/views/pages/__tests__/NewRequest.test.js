import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider } from '@material-ui/core';
import { MemoryRouter } from 'react-router-dom';
import NewRequest from '../NewRequest/NewRequest';

test('New Request Page renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <MemoryRouter>
                <NewRequest />
            </MemoryRouter>
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
