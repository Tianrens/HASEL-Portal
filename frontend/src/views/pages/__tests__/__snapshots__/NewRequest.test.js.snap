// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Request Page renders properly 1`] = `
<div
  className="container"
>
  <h1
    className="header"
  >
    You'll need approval to access a workstation.
  </h1>
  <form
    autoComplete="off"
    className="form"
    onSubmit={[Function]}
  >
    <p
      className="inputTitles"
    >
      Reasoning/Comments (optional)
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        onClick={[Function]}
      >
        <textarea
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          rows={4}
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-2"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <p
      className="inputTitles"
    >
      Workstation (may be overridden)
    </p>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth"
      onClick={[Function]}
      required={true}
    >
      <div
        aria-haspopup="listbox"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={true}
        tabIndex={-1}
        value=""
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-2"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
    <div
      className="buttonContainer"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained button undefined"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Request Access
        </span>
      </button>
    </div>
  </form>
</div>
`;
